ddtrace/tracer.py

```diff
@@ -719,8 +719,8 @@ def _start_span(
             span._metrics[PID] = self._pid
 
         # Apply default global tags.
-        if self._tags:
-            span.set_tags(self._tags)
+        if config.tags:
```

Comment:
Since `config.tags` is now a `_ConfigItem` this value will be responsive to global config API updates when it previous wasn't:


```python
from ddtrace import config, tracer

config.tags = { "a": "b" }

with tracer.trace("") as span:
    pass
```

Previously, `span.get_tag("a")` would return `None`, now it will return `"b"`.

This comes at a slight performance hit as there is another level of indirection for the `getattr` on the global config. We could mitigate this easily enough by keeping the tags reference on the tracer... which we should just do... Gonna make that change lol